docs/standards/smart-contracts/utils/lsp2-json-schema-utils.md

```diff
@@ -0,0 +1,347 @@
+
+#### Return Values:
+
+| Name     | Type    | Description     |
+| :------- | :------ | :-------------- |
+| `result` | bytes32 | JSON URL Value. |
+
+### generateASSETURLValue
+
+```solidity
+function generateASSETURLValue(
+    string memory hashFunction,
+    string memory assetBytes,
+    string memory url
+) internal pure returns (bytes memory key);
+```
+
+Generate a ASSETURL valueContent.
+
+#### Parameters:
+
+| Name           | Type   | Description                              |
+| :------------- | :----- | :--------------------------------------- |
+| `hashFunction` | string | The function used to hash the JSON file. |
+| `assetBytes`   | string | Bytes value of the JSON file.            |
+| `url`          | string | The URL where the JSON file is hosted.   |
+
+#### Return Values:
+
+| Name     | Type    | Description      |
+| :------- | :------ | :--------------- |
+| `result` | bytes32 | ASSET URL Value. |
+
+### isEncodedArray
+
+```solidity
+function isEncodedArray(
+    bytes memory data
+) internal pure returns (bool);
+```
+
+Verifing if `data` is an encoded array
+
+#### Parameters:
+
+| Name   | Type  | Description                       |
+| :----- | :---- | :-------------------------------- |
+| `data` | bytes | The value that is to be verified. |
+
+### isEncodedArrayOfAddresses
+
+```solidity
+function isEncodedArrayOfAddresses(
+    bytes memory data
+) internal pure returns (bool);
+```
+
+Verifing if `data` is an encoded array of addresses (address[])
+
+#### Parameters:
+
+| Name   | Type  | Description                       |
+| :----- | :---- | :-------------------------------- |
+| `data` | bytes | The value that is to be verified. |
+
+### isBytes4EncodedArray
+
+```solidity
+function isBytes4EncodedArray(
+    bytes memory data
+) internal pure returns (bool);
+```
+
+Verify that `data` is an array of bytes4 (bytes4[]) encoded according to the Solidity ABI specs.
+
+#### Parameters:
+
+| Name   | Type  | Description                       |
+| :----- | :---- | :-------------------------------- |
+| `data` | bytes | The value that is to be verified. |
+
+### isCompactBytesArray
+
+```solidity
+function isCompactBytesArray(
+    bytes memory compactBytesArray
+) internal pure returns (bool);
+```
+
+Verify the validity of the `compactBytesArray` according to LSP2.
+
+#### Parameters:
+
+| Name                | Type  | Description                       |
+| :------------------ | :---- | :-------------------------------- |
+| `compactBytesArray` | bytes | The value that is to be verified. |
+
+### uncheckedIncrement
+
+```solidity
+uncheckedIncrement(
+    uint256 i
+) internal pure returns (uint256);
+```
+
+Will return unchecked incremented uint256.
+Can be used to save gas when iterating over loops.
+
+## References
+
+- [Solidity implementations (GitHub)](https://github.com/lukso-network/lsp-universalprofile-smart-contracts/tree/develop/contracts)
```

Comment:
Not really related to your changes, but this link is quite old, we can replace it with the new one but wonder if we should keep it or remove it